pkg/asset/machines/master.go

```diff
@@ -606,3 +610,37 @@ func IsMachineManifest(file *asset.File) bool {
 		return matched
 	}
 }
+
+func generateAsset(elements interface{}, fileName string) (assets []*asset.File, err error) {
```

Comment:
This change's main idea looks good to me to avoid repetitions. But I think, it would be better to have 2 functions, `generateAssetForHost`, `generateAssetForSecret` instead this generic function, thanks to that, we can pass concrete types instead interface{} and we can benefit from compile time checks.

Reply:
Thanks @ardaguclu good idea